packages/devextreme/js/__internal/grids/grid_core/selection/m_selection.ts

```diff
@@ -299,6 +299,7 @@ export class SelectionController extends modules.Controller {
   private _createSelection() {
     const options = this._getSelectionConfig();
 
+    // @ts-expect-error TKey
     return new Selection(options);
```

Comment:
Let's change this `ts-expect-error` a little bit
The problem that `this._getSelectionConfig` returned object missed some required properties from the 
`Selection` -> `DefaultOptions` type

So, from my perspective better to highlight this type inconsistency and move `ts-expect-error` to the `options` line:
```suggestion
    // @ts-expect-error inconsistent types with SelectionOptions and this._getSelectionConfig
    const options: SelectionOptions = this._getSelectionConfig();
    
    return new Selection(options);
```

As a further TS improvement steps in the future, from our side we will try to make this types consistent and remove this `ts-expect-error` :)

Reply:
Right now SelectionOptions, that are passed to the constructor are all Partial and the error is not about missing properies.
It is about `selectedKeys` property, which has type `unknown[] | undefined`, that's because type of TKey is `unknown`. and the expected type of TKey is `string | number`

Also, default props should not all be required for passing here. Because they will fallback to default ones. There may be some required props, that are not in default (this will be established in refactoring issue).

Reply:
Ok, let's leave it as is for now then :)